src/mender-update/cli/actions.cpp

```diff
@@ -150,83 +152,75 @@ error::Error ShowProvidesAction::Execute(context::MenderContext &main_context) {
+		} else if (
+			ResultContains(result.result, Result::Downloaded | Result::RolledBack)
+			or ResultContains(result.result, Result::Installed | Result::RolledBack)) {
+			cout << "Installation failed. Rolled back modifications." << endl;
+		} else if (ResultContains(result.result, Result::RollbackFailed)) {
+			cout << "Rollback failed. System may be in an inconsistent state." << endl;
+		} else if (ResultContains(result.result, Result::CleanupFailed)) {
+			cout << "Installed, but one or more post-commit steps failed." << endl;
+		} else {
+			stringstream ss;
+			ss << std::hex << static_cast<int>(result.result);
+			log::Error("Unexpected result value: 0x" + ss.str());
+		}
+	} else if (ResultIs(result.result, Result::RolledBack)) {
+		cout << "Rolled back." << endl;
+	} else if (ResultIs(result.result, Result::NoUpdateInProgress)) {
+		cout << "No update in progress." << endl;
+	} else if (ResultIs(result.result, Result::Downloaded)) {
+		cout << "Streamed to storage, but not installed/enabled." << endl;
+	} else if (ResultContains(result.result, Result::Installed | Result::Committed)) {
 		cout << "Installed and committed." << endl;
-		break;
-	case standalone::Result::Committed:
-		cout << "Committed." << endl;
-		break;
-	case standalone::Result::Installed:
-	case standalone::Result::InstalledRebootRequired:
+	} else if (ResultContains(result.result, Result::Installed)) {
 		cout << "Installed, but not committed." << endl;
 		cout << "Use 'commit' to update, or 'rollback' to roll back the update." << endl;
-		break;
-	case standalone::Result::InstalledButFailedInPostCommit:
-		cout << "Installed, but one or more post-commit steps failed." << endl;
-		break;
-	case standalone::Result::NoUpdateInProgress:
-		cout << "No update in progress." << endl;
-		break;
-	case standalone::Result::FailedNothingDone:
-		cout << "Installation failed. System not modified." << endl;
-		break;
-	case standalone::Result::RolledBack:
-		cout << "Rolled back." << endl;
-		break;
-	case standalone::Result::NoRollback:
-		cout << "Update Module does not support rollback." << endl;
-		break;
-	case standalone::Result::RollbackFailed:
-		cout << "Rollback failed. System may be in an inconsistent state." << endl;
-		break;
-	case standalone::Result::FailedAndRolledBack:
-		cout << "Installation failed. Rolled back modifications." << endl;
-		break;
-	case standalone::Result::FailedAndNoRollback:
-		cout
-			<< "Installation failed, and Update Module does not support rollback. System may be in an inconsistent state."
-			<< endl;
-		break;
-	case standalone::Result::FailedAndRollbackFailed:
-		cout
-			<< "Installation failed, and rollback also failed. System may be in an inconsistent state."
-			<< endl;
-		break;
+	} else if (ResultContains(result.result, Result::Committed)) {
+		cout << "Committed." << endl;
+	} else {
+		stringstream ss;
+		ss << std::hex << static_cast<int>(result.result);
+		log::Error("Unexpected result value: 0x" + ss.str());
 	}
```

Comment:
TBH, this whole if/else if cascade was reading much better with the `switch` statement.

I couldn't spot anything wrong, though. It seems to be doing the same as before

Reply:
Using a switch statement seems infeasible with the new possibilities for starting and stopping, because there are so many combinations, and adding new results need to take existing results into account, hence the bit flags. However, I do believe I have considerably improved the readability now, using a combination of what we talked about yesterday (split some things which were previously sentences into multiple lines), and some new bit flags. Check it out!